### PR TITLE
Load balancer map removal v2

### PR DESCRIPTION
**Pull request by maartenvanderhoef**

### CODE REVIEW

File: modules/iam/main.tf [hunk 1]
```diff
@@ -73,7 +73,7 @@ data "aws_iam_policy_document" "ssm_permissions" {
 resource "aws_iam_role_policy" "ssm_permissions" {
   count  = "${(var.create && var.ssm_enabled) ? 1 : 0 }"
   name   = "${var.name}-ssm-permissions"
-  role   = "${aws_iam_role.ecs_tasks_role.id)}"
```
Comment (Jamie-BitFlight) [inline modules/iam/main.tf]: oops!

File: modules/iam/main.tf [hunk 2]
```diff
@@ -22,21 +22,21 @@ data "aws_iam_policy_document" "ecs_task_assume_role" {
 resource "aws_iam_role" "ecs_task_execution_role" {
   count              = "${(var.create && var.fargate_enabled ) ? 1 : 0 }"
   name               = "${var.name}-ecs-task-execution_role"
-  assume_role_policy = "${join(data.aws_iam_policy_document.ecs_task_assume_role.*.json)}"
```
Comment (Jamie-BitFlight) [inline modules/iam/main.tf]: oops

Comment (maartenvanderhoef) [reply modules/iam/main.tf]: I agree, join with only a single argument should just join with "" ;-)

**Comment by maartenvanderhoef** [comment]
> It looks good. But if the dependancies null_resource is removed does it still solve the issue with count cannot be computed when the alb is created by another module?

I'd like to give that a shot later, I don't see why that wouldn't work now. I'm thinking to first merge this into your branch, let Jon take a look at your original (now) changed PR and then merge to master so the rest can continue with the new variable structure.

**Pull request merged**